src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedPhysicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,149 @@
+package se.liu.ida.hefquin.engine.queryplan.utils;
+
+import java.io.PrintStream;
+
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpBGPAdd;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalPlanVisitor;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalPlanWalker;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.*;
+import se.liu.ida.hefquin.engine.utils.IndentingPrintStream;
+
+public class TextBasedPhysicalPlanPrinterImpl2 implements PhysicalPlanPrinter
+{
+	// The string represents '|   '.
+	private static String levelIndentBase = "\u2502   ";
+	// The string represents '├── '.
+	private static String nonLastChildIndentBase = "\u251C\u2500\u2500 ";
+	// The string represents '└── '.
+	private static String lastChildIndentBase = "\u2514\u2500\u2500 ";
+	private static String spaceBase = "    ";
+		
+	@Override
+	public void print( final PhysicalPlan plan, final PrintStream out ) {
+		planWalk(plan, 0, 0, 1, out, "");
+		out.flush();
+	}
+
+	public String getIndentLevelString(final int planNumber, final int planLevel, final int numberOfSiblings, final String upperRootOpIndentString) {
```

Comment:
This functions seems to be something that can be moved to a common base class as well.